File: src/shared/abstractions/Localizer.ts
import { Langu, Messages } from "../../application/resources";
import { ILocalizer } from "./ILocalizer";

export class Localizer implements ILocalizer {
    definedLangu: Langu
    constructor(langu: Langu) {
        this.definedLangu = langu ?? Langu.EN;
    }

    m(key: string): string {
        const m = Messages.get(key)
        if (m == null) {
            return key + ": " + key
        } else {
            return key + ": " + m.find(e => e.langu == this.definedLangu)?.message
        }
    }

}

File: toolkit/localizer/localizer.go
package localizer

const EN string = "en"
const ES string = "es"

type ILocalizer interface {
	Localize(id string) string
	SetLanguage(langu string)
}

type localizer struct {
	messages map[string]map[string]string
	langu    string
}

func NewLocalizer(langu string, messageMap map[string]map[string]string) *localizer {
	return &localizer{
		messages: messageMap,
		langu:    langu,
	}
}

func (l *localizer) SetLanguage(langu string) {
	l.langu = langu
}

func (l localizer) Localize(id string) string {
	mess := l.messages[id][l.langu]
	if mess == "" {
		mess = l.messages[id][EN]
	}
	if mess == "" {
		mess = id
	}

	return mess
}
